OrbitQt/orbitmainwindow.cpp

```diff
@@ -217,6 +223,107 @@ OrbitMainWindow::OrbitMainWindow(QApplication* a_App,
   GOrbitApp->PostInit();
 }
 
+void SetFontSize(QWidget* widget, uint32_t font_size) {
+  QFont font = widget->font();
+  font.setPointSize(font_size);
+  widget->setFont(font);
+}
+
+QWidget* CreateSpacer(QWidget* parent) {
+  QLabel* spacer = new QLabel(parent);
+  spacer->setText("    ");
+  return spacer;
+}
+
+QAction* CreateDummyAction(const QIcon& icon, QObject* parent) {
+  QAction* action = new QAction(icon, "", parent);
+  action->setDisabled(true);
+  return action;
+}
+
+QIcon GetIcon(const std::string& icon_name) {
+  return QIcon(Path::JoinPath({Path::GetIconsPath(), icon_name}).c_str());
+}
+
+void OrbitMainWindow::SetupCaptureToolbar() {
+  // Sizes.
+  uint32_t kFontSize = 10;
+  uint32_t kIconSize = 30;
+  QToolBar* toolbar = ui->capture_toolbar;
+  toolbar->setIconSize(QSize(kIconSize, kIconSize));
+
+  // Create icons.
+  QIcon icon_start = GetIcon("outline_play_arrow_white_48dp.png");
+  QIcon icon_stop = GetIcon("outline_stop_white_48dp.png");
+  QIcon icon_clear = GetIcon("outline_clear_white_48dp.png");
+  QIcon icon_open = GetIcon("outline_folder_white_48dp.png");
+  QIcon icon_save = GetIcon("outline_save_alt_white_48dp.png");
+  QIcon icon_help = GetIcon("outline_help_outline_white_48dp.png");
+  QIcon icon_feedback = GetIcon("outline_feedback_white_48dp.png");
+  QIcon icon_search = GetIcon("outline_search_white_48dp.png");
+  QIcon icon_filter = GetIcon("outline_filter_list_white_48dp.png");
+  QIcon icon_timer = GetIcon("outline_access_time_white_48dp.png");
+
+  // Set action icons.
+  ui->actionStart_Capture->setIcon(icon_start);
+  ui->actionStop_Capture->setIcon(icon_stop);
+  ui->actionClear_Capture->setIcon(icon_clear);
+  ui->actionOpen_Capture->setIcon(icon_open);
+  ui->actionSave_Capture->setIcon(icon_save);
+  ui->actionHelp->setIcon(icon_help);
+  ui->actionFeedback->setIcon(icon_feedback);
+  ui->actionFilter_Functions->setIcon(icon_search);
+  ui->actionFilter_Tracks->setIcon(icon_filter);
+
+  // Add actions.
+  toolbar->addAction(ui->actionStart_Capture);
+  toolbar->addAction(ui->actionStop_Capture);
+  toolbar->addAction(ui->actionClear_Capture);
+  toolbar->addAction(ui->actionOpen_Capture);
+  toolbar->addAction(ui->actionSave_Capture);
+  toolbar->addAction(ui->actionHelp);
+  toolbar->addAction(ui->actionFeedback);
+
+  // Filter tracks.
+  toolbar->addWidget(CreateSpacer(toolbar));
+  toolbar->addAction(CreateDummyAction(icon_filter, toolbar));
```

Comment:
> Just from the screenshot: What about taking it out of the "capture" tab and positioning it directly under the menu bar like toolbars normally are? This way it would always use the full width of the window (I don't think it's a problem if we lose a tiny bit of vertical space on the right side). Otherwise I have a feeling that when the capture tab is not that wide it would take several lines. 

I really want to minimize vertical space taken up by UI. I initially tried having the toolbar at the top level, and didn't like how it looked but more importantly I didn't think it made sense for all the actions to be available globally.

> Also some buttons don't only apply to the capture view, for example load, save, feedback.

Those buttons all apply to the capture view, did you mean not "only" apply to the capture view? Some of those actions are already present in the main menu, so they are accessible from anywhere. Inversely, having the toolbar at the top level would expose functionality that is not at all pertinent to any other tabs than the capture view (start/stop capture without visual feedback is a risk; filter tracks and the timer label wouldn't make sense either in any other view.

> Can some of this code be replaced with definitions in the .ui file (e.g.: adding the actions and the other widgets to the toolbar)? That's what is done for the menu.

That's how it was initially done but I found that as soon as you need to do something a little bit special (adding a spaces between elements for example) then you'd have to add some code.  Also, in Qt Creator, you can't add a QLineEdit to a toolbar, this needs to be done in code, or possibly by editing the xml manually which I'm not a fan of.  I found that we have way more control in code.

> Isn't it possible to add an icon to the toolbar directly?

You can't add a QIcon directly to a toolbar.  I tried adding a QPushButton instead of the action, but the look is different.